internal/pkg/util/env/clean.go

```diff
@@ -15,76 +15,100 @@ import (
 const (
 	// DefaultPath defines default value for PATH environment variable.
 	DefaultPath = "/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin"
+	// SingularityPrefix defines the environment variable prefix SINGULARITY_.
+	SingularityPrefix = "SINGULARITY_"
+	// SingularityEnvPrefix defines the environment variable prefix SINGULARITYENV_.
+	SingularityEnvPrefix = "SINGULARITYENV_"
 )
 
 var alwaysPassKeys = map[string]struct{}{
-	"term":        {},
+	"TERM":        {},
 	"http_proxy":  {},
+	"HTTP_PROXY":  {},
 	"https_proxy": {},
+	"HTTPS_PROXY": {},
 	"no_proxy":    {},
+	"NO_PROXY":    {},
 	"all_proxy":   {},
+	"ALL_PROXY":   {},
 	"ftp_proxy":   {},
+	"FTP_PROXY":   {},
 }
 
-var alwaysOmitKeys = map[string]struct{}{
-	"path":            {},
-	"ld_library_path": {},
+// boolean value defines if the variable could be overridden
+// with the SINGULARITYENV_ variant.
+var alwaysOmitKeys = map[string]bool{
+	"HOME":            false,
+	"PATH":            false,
+	"LD_LIBRARY_PATH": true,
 }
 
 // SetContainerEnv cleans environment variables before running the container.
 func SetContainerEnv(g *generate.Generator, hostEnvs []string, cleanEnv bool, homeDest string) {
+	singEnvKeys := make(map[string]string)
+
+	// allow override with SINGULARITYENV_LANG
+	if cleanEnv {
+		g.AddProcessEnv("LANG", "C")
```

Comment:
Is it always reasonable to force `LANG` to `C` when  a UTF-8 locale is the default almost everywhere now?

Reply:
> UTF-8 locale is the default almost everywhere now

I don't know, it was always done like that, just moved the code before looping over environment variables to allow `SINGULARITYENV_LANG` to be passed in conjunction with `-e` like any other variables that a user would like to forward to container when using clean environment.

I think `C` was chosen because many images don't have locales (especially from docker) installed and C is de-facto the default here but in the meantime this is a bit weird to do that only with a cleaned environment and not all the time :

> Also, could you add a note about the behaviour change to the CHANGELOG.md?

Yep, do we integrate it for 3.5.3 ?

Reply:
Okay - let's leave it as C here for now.

> Yep, do we integrate it for 3.5.3 ?

I think elsewhere I promised we wouldn't change behaviour in 3.5 and that we would look at addressing all the inconsistency including docker env var stuff: https://github.com/sylabs/singularity/issues/4827

I still think it'd be better to do this and make all behaviour changes for consistency in a `3.6.0` release if we can. Do you think that is reasonable?

Reply:
Sounds good